app/models/user.rb

```diff
@@ -17,6 +17,18 @@ def authorized_to?(access_request)
     end
   end
 
+  def inventory_groups
+    rbac_permissions.map do |permission|
+      next if permission.permission != Rbac::INVENTORY_GROUPS_READ
+      return Rbac::GLOBAL_RESOURCE_PERMISSION if permission.resource_definitions == []
+
+      filter = permission.resource_definitions.first[:attributeFilter]
+      next unless Rbac.supported_operation?(filter)
+
+      Rbac.operation_value(filter)
+    end.flatten.compact
```

Comment:
Code looks good, however, it should have some comments to explain things. Especially the jumps that need explanation.